contrib/golang/filters/http/source/common/dso/dso.cc

```diff
@@ -0,0 +1,161 @@
+#include "contrib/golang/filters/http/source/common/dso/dso.h"
+
+#include "source/common/common/assert.h"
+
+namespace Envoy {
+namespace Dso {
+
+std::map<std::string, DsoInstance*> DsoInstanceManager::dso_map_ = {};
+absl::Mutex DsoInstanceManager::mutex_ = {};
+
+bool DsoInstanceManager::pub(std::string dso_id, std::string dso_name) {
+  if (getDsoInstanceByID(dso_id) != nullptr) {
+    ENVOY_LOG_MISC(warn, "pub {} {} dso instance failed: already pub.", dso_id, dso_name);
+    return true;
+  }
+
+  absl::WriterMutexLock lock(&DsoInstanceManager::mutex_);
+
+  DsoInstance* dso = new DsoInstance(dso_name);
```

Comment:
use unique_ptr here and elsewhere. Please no raw memory management.

Reply:
Done